<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="./img/favicon.ico">

	<title>My Docs</title>

        <link href="./css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="./css/font-awesome-4.5.0.css" rel="stylesheet">
        <link href="./css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="./css/highlight.css">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body class="homepage">

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            

            <!-- Main title -->
            <a class="navbar-brand" href=".">My Docs</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            

            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#introduction-to-galaxy">Introduction to Galaxy</a></li>
        
            <li><a href="#background">Background</a></li>
        
            <li><a href="#learning-objectives">Learning Objectives</a></li>
        
            <li><a href="#section-1-preparation">Section 1: Preparation.</a></li>
        
            <li><a href="#section-2-getting-data-into-galaxy">Section 2: Getting data into Galaxy</a></li>
        
            <li><a href="#section-3-play-with-the-tools">Section 3: Play with the tools</a></li>
        
            <li><a href="#what-now">What now?</a></li>
        
    
        <li class="main "><a href="#galaxy-workflows">Galaxy Workflows</a></li>
        
            <li><a href="#background_1">Background</a></li>
        
            <li><a href="#section-1-preparation_1">Section 1: Preparation.</a></li>
        
            <li><a href="#section-2-create-and-run-a-workflow">Section 2: Create and run a workflow.</a></li>
        
            <li><a href="#what-now_1">What now?</a></li>
        
    
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<h1 id="introduction-to-galaxy">Introduction to Galaxy</h1>
<p>The tutorial is the modified version which is written and maintained by <a href="mailto:simon.gladman@unimelb.edu.au">Simon
Gladman</a> - VLSCI</p>
<h2 id="background">Background</h2>
<p>Galaxy is a web based analysis and workflow platform designed for
biologists to analyse their own data. It comes with most of the popular
bioinformatics tools already installed and ready for use. There are many
Galaxy servers around the world and some are tailored with specific
toolsets and reference data for analysis of human genomics, microbial
genomics, proteomics etc.</p>
<p>There are some introductory slides available
<a href="https://docs.google.com/presentation/d/1dzHagGkswjH7MOZ7OACVXGU-riBs33K3J5lWpnCpPhs/pub?start=false&amp;loop=false&amp;slide=id.g537781ff1_2_18">here</a>.</p>
<p>Basically, the Galaxy interface is separated into 3 parts. The tool list
on the left, the viewing pane in the middle and the analysis and data
history on the right. We will be looking at all 3 parts in this
tutorial.</p>
<p><img alt="Galaxy interface components diagram" src="./images/Galaxy_components.png" /></p>
<p>This workshop/tutorial will familiarize you with the Galaxy interface.
It will cover the following topics:</p>
<ul>
<li>Logging in to the server</li>
<li>Getting data into galaxy</li>
<li>How to access the tools</li>
<li>Using to use some common tools</li>
</ul>
<hr />
<h2 id="learning-objectives">Learning Objectives</h2>
<p>At the end of this tutorial you should:</p>
<p>Be able to upload data to a Galaxy server from:
    -   A file on your local computer
    -   A file on a remote datastore with an accessible URL.
    -   A file UCSC genome browser <strong>missing</strong></p>
<p>Be able use tools in Galaxy by:
    -   Accessing the tool via the tool menu
    -   Using the tool interface to run the particular tool
    -   Viewing/accessing the tool output.</p>
<hr />
<h2 id="section-1-preparation">Section 1: Preparation.</h2>
<p>The purpose of this section is to get you to log in to the server..</p>
<ol>
<li>
<p>Go to <a href="https://galaxy.mdc-berlin.net/galaxy/">MDC galaxy production sever</a></p>
</li>
<li>
<p>If you have previously registered on this server just log in:</p>
<ul>
<li>Enter your MDC user name</li>
<li>Enter your password</li>
<li>Click <strong>OK</strong></li>
</ul>
</li>
</ol>
<hr />
<h2 id="section-2-getting-data-into-galaxy">Section 2: Getting data into Galaxy</h2>
<p>There are 2 main ways to get your data into Galaxy. We will use each of
these methods for 3 files and then work use those 3 files for the rest
of the workshop.</p>
<ul>
<li>
<p>Start a new history for this workshop. To do this:</p>
<ul>
<li>Click on the history menu button (the
    <img alt="" src="./images/Galaxy-menu.png" /> icon)
    at the top of the Histories panel.</li>
<li>Select <strong>Create New</strong></li>
</ul>
</li>
</ul>
<p>It is important to note that Galaxy has the concept of "File Type" built
in. This means that each file stored needs to have its type described to
Galaxy as it is being made available. Examples of file types are: text,
fasta, fastq, vcf, GFF, Genbank, tabular etc.</p>
<p>We will tell Galaxy what type of file each one is as we upload it.</p>
<h3 id="method-1-upload-a-file-from-your-own-computer">Method 1: Upload a file from your own computer</h3>
<p>With this method you can get most of the files on your own computer into
Galaxy. (there is a size limit)</p>
<ul>
<li>
<p>Download the following file to your computer:
    <em>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/galaxy101/Contig_stats.txt.gz</em></p>
<ul>
<li>From the Galaxy tool panel, click on <strong>Get Data -&gt; Upload
    File</strong></li>
<li>Click the <strong>Choose File</strong> button</li>
<li>Find and select the <em>Contig_stats.txt.gz</em> file you downloaded
    and click <strong>Open</strong></li>
<li>Set the "file format" to <em>tabular</em></li>
<li>Click the <strong>Start</strong> button</li>
<li>Once the progress bar reaches 100%, click the <strong>Close</strong> button</li>
</ul>
</li>
</ul>
<p>The file will now upload to your current history.</p>
<h3 id="method-2-upload-a-file-from-a-url">Method 2: Upload a file from a URL</h3>
<p>If a file exists on a web resource somewhere and you know its URL
(Unique resource location - a web address) you can directly load it into
Galaxy.</p>
<ul>
<li>
<p>From the tool panel, click on <strong>Get Data -&gt; Upload File</strong></p>
<ul>
<li>Click on the <strong>Paste/Fetch Data</strong> button</li>
<li>Copy and paste the following web address into the URL/Text box:
    <em>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/COMP90014/Assignment1/bacterial_std_err_1.fastq.gz</em></li>
<li>Set the file format to <em>fastqsanger</em> (not fastqcsanger)</li>
<li>Click <strong>Start</strong></li>
<li>Once the progress bar has reached 100%, click <strong>Close</strong></li>
</ul>
</li>
</ul>
<p>Note that Galaxy is smart enough to recognize that this is a compressed
file and so it will uncompress it as it loads it.</p>
<h3 id="method-2-again-get-data-from-a-public-url">Method 2 (again): Get data from a public URL</h3>
<p>Now we are going to upload another file from the remote data source.</p>
<p>Repeat the above for:
<em>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/MRSA0252.fna</em></p>
<p>Note that this file is a <em>fasta</em> file and not a <em>fastqsanger</em> file.</p>
<p>The DNA sequence of <em>Staphlococcus aureus MRSA252</em> will be loaded into
your history as a fasta file.</p>
<h3 id="the-data">The data</h3>
<p>Though we aren't going to focus on the contents of these files and what
they mean from a bioinformatics standpoint, here is a brief description
of each one.</p>
<ul>
<li>
<p><em>Contigs_stats.txt</em></p>
<ul>
<li>this file contains a table of summary data from a de novo genome
    assembly (the process of attempting to recover the full genome
    of an organism from the short read sequences produced by most
    DNA sequencing machines. )</li>
<li>The columns contain a lot of information but the ones we will be
    using indicate the amount of data (or coverage) that went into
    making up each piece of the final assembly.</li>
</ul>
</li>
</ul>
<!-- -->

<ul>
<li>
<p><em>bacterial_std_err_1.fastq.gz</em></p>
<ul>
<li>This file contains sequence reads as they would come off an
    Illunina sequencing machine. They are in
    <a href="https://en.wikipedia.org/wiki/FASTQ_format">fastq</a> format.</li>
</ul>
</li>
</ul>
<!-- -->

<ul>
<li>
<p><em>MRSA0252.fna</em></p>
<ul>
<li>This file contains the genome sequence of <em>Staphylococcus aureus
    MRSA252</em>. It is in
    <a href="https://en.wikipedia.org/wiki/FASTA_format">fasta</a> format.</li>
</ul>
</li>
</ul>
<hr />
<h2 id="section-3-play-with-the-tools">Section 3: Play with the tools</h2>
<p>The purpose of this section is to get you used to using the available
tools in Galaxy and point out some of the more basic manipulation tools.</p>
<p>Firstly however, you’ll notice that two of the files have very long and
confusing names. So we might want to change them. To do this we need to
“edit” the file. So:</p>
<ol>
<li>Click on the
    <img alt="" src="./images/Galaxy-edit.png" />
    icon (edit) next to the file in the history called:
    <em>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/COMP90014/Assignment1/bacterial_std_err_1.fastq</em></li>
<li>In the "Name" text box, give it a new name. Call it: <em>Typical Fastq
    File</em></li>
<li>Click the <strong>Save</strong> button.</li>
</ol>
<p>Repeat the process for the MRSA252 fasta file. Rename it to
<em>MRSA252.fna</em></p>
<p>Now that’s better. There was a lot of other functionality hidden behind
that edit
(<img alt="" src="./images/Galaxy-edit.png" />)
icon. You can change a file’s data type, convert its format and many
other things. Feel free to play around with them at a later date.</p>
<p>Ok, back to the tools..</p>
<h3 id="example-1-histogram-and-summary-statistics">Example 1: Histogram and summary statistics</h3>
<p>The first thing we are going to do is produce a histogram of contig read
coverage depths and calculate the summary statistics from the
Contig_stats.txt file. To do this we need to cut out a couple of
columns, remove a line and then produce a histogram. This will introduce
some of the text manipulation tools.</p>
<p>Click on the
<img alt="" src="./images/Galaxy-view.png" />
icon of the <em>Contig_stats.txt</em> file to have a look at it. Note that
there are 18 columns in this file. We want column 1 and column 6. To do
this:</p>
<p><strong>1. Cut out column 1 and column 6.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Text Manipulation -&gt; Cut</strong> and
    set the following:</li>
<li>Set "Cut Columns" to: <em>c1,c6</em></li>
<li>"Delimited by": <em>Tab</em></li>
<li>"Cut from": <em>Contig_stats.txt</em></li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>Examine the new file by clicking on it’s
<img alt="" src="./images/Galaxy-view.png" />
icon. We now have 2 columns instead of the 18 in the original file.</p>
<p><strong>2. Remove the Header lines of the new file.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Text Manipulation -&gt; Remove
    beginning</strong> and set the following:</li>
<li>"Remove First": <em>1</em></li>
<li>"from": <em>Cut on data1</em></li>
<li>click <strong>Execute</strong></li>
</ul>
<p>Note the the new file is the same as the previous one without the header
line.</p>
<p><strong>3. Make a histogram.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Graph/Display Data -&gt; Histogram</strong>
    and set the following:</li>
<li>"Dataset": <em>Remove beginning on Data X</em></li>
<li>"Numerical column for X axis": <em>c2</em></li>
<li>"Number of breaks": <em>25</em></li>
<li>"Plot title": <em>Histogram of Contig Coverage</em></li>
<li>"Label for X axis": <em>Coverage depth</em></li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>Click on the
<img alt="" src="./images/Galaxy-view.png" />
icon of the histogram to have a look at it. Note there are a few peaks..
Maybe these correspond to single, double and triple copy number of these
contigs.</p>
<p><strong>4. Calculate summary statistics for contig coverage depth.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Statistics -&gt; Summary
    Statisitics</strong> and set the following:</li>
<li>"Summary statistics on": <em>Remove beginning on Data X</em></li>
<li>"Column or expression": <em>c2</em></li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>You’ll note that the summary statistics tool failed and is red in the
history. There was an error! If you click on the filename, and then the
bug
<img alt="" src="./images/Galaxy-bug.png" />
symbol, it will tell you what went wrong. (There is a missing python
library.) At this point, you would normally contact your Galaxy server
administrator.</p>
<h3 id="example-2-convert-fastq-to-fasta">Example 2: Convert Fastq to Fasta</h3>
<p>This shows how to convert a fastq file to a fasta file. The tool creates
a new file with the converted data.</p>
<p><strong>Converter tool</strong></p>
<ul>
<li>From the tool panel, click on <strong>Convert Formats -&gt; FASTQ to
    FASTA</strong> and set the following:</li>
<li>"FASTQ file to convert": <em>Typical Fastq File</em></li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>This will have created a new Fasta file called FASTQ to FASTA on data 2.</p>
<h3 id="example-3-find-ribosomal-rna-features-in-a-dna-sequence">Example 3: Find Ribosomal RNA Features in a DNA Sequence</h3>
<p>This example shows how to use a tool called “barrnap” to search for
rRNAs in a DNA sequence.</p>
<p><strong>1. Find all of the ribosomal RNA's in a sequence</strong></p>
<ul>
<li>From the tool panel, click on <strong>Annotation -&gt; barrnap</strong> and set
    the following:</li>
<li>"Fasta file": MRSA252.fna</li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>A new file called <em>barrnap on data 3</em> will be produced. It is a gff3
file. (This stands for genome feature format - version 3. It is a file
format for describing features contained by a DNA sequence.) Change it’s
name to something more appropriate (click on the
<img alt="" src="./images/Galaxy-edit.png" />
icon.) There is also a STDERR output file from this tool - just ignore
this one.</p>
<p>Now lets say you only want the lines of the file for the 23S rRNA
annotations. We can do this using a Filter tool.</p>
<p><strong>2. Filter the annotations to get the 23S RNAs</strong></p>
<ul>
<li>From the tool panel, click on <strong>Filter and Sort -&gt; Select</strong> and
    set the following:</li>
<li>"Select lines from": (whatever you called the barrnap gff3 output)</li>
<li>"the pattern": <em>23S</em> (this will look for all the lines in the file
    that contain “23S”)</li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>Now you have a gff3 file with just the 23S annotations!</p>
<h2 id="what-now">What now?</h2>
<p>Remember how we started a new history at the beginning? If you want to
see any of your old histories, click on the history menu button
<img alt="" src="./images/Galaxy-menu.png" />
at the top of the histories panel and then select “Saved Histories.”
This will give you a list of all the histories you have worked on in
this Galaxy server.</p>
<p>That's it. You now know a bit about the Galaxy interface and how to load
data, run tools and view their outputs. For more tutorials, see
<a href="http://genome.edu.au/learn">http://genome.edu.au/learn</a></p>
<hr />
<ul>
<li><a href="http://vlsci.github.io/lscc_docs/tutorials/galaxy-workflows/galaxy-workflows/#galaxy-workflows">Galaxy
    Workflows</a></li>
<li><a href="http://vlsci.github.io/lscc_docs/tutorials/galaxy-workflows/galaxy-workflows/#background">Background</a></li>
<li><a href="http://vlsci.github.io/lscc_docs/tutorials/galaxy-workflows/galaxy-workflows/#section-1-preparation">Section
    1: Preparation.</a></li>
<li><a href="http://vlsci.github.io/lscc_docs/tutorials/galaxy-workflows/galaxy-workflows/#section-2-create-and-run-a-workflow">Section 2: Create and run
    a workflow.</a></li>
<li><a href="http://vlsci.github.io/lscc_docs/tutorials/galaxy-workflows/galaxy-workflows/#what-now">What
    now?</a></li>
</ul>
<h1 id="galaxy-workflows">Galaxy Workflows</h1>
<h2 id="background_1">Background</h2>
<p>This workshop/tutorial will familiarise you with the Galaxy workflow
engine. It will cover the following topics:</p>
<ul>
<li>Logging in to the server</li>
<li>How to construct and use a workflow by various methods</li>
<li>How to share a workflow</li>
</ul>
<hr />
<h2 id="section-1-preparation_1">Section 1: Preparation.</h2>
<p>The purpose of this section is to get you to log in to the server.. For
this workshop, you can use a Galaxy server that you have created by
following the steps outlined in: <a href="">Launch a GVL Galaxy Instance</a> or you
can use the public <a href="http://galaxy-tut.genome.edu.au/">Galaxy-tut</a></p>
<p><strong>Go to Galaxy URL of your server in Firefox or Chrome (your choice,
please don't use IE or Safari.)</strong></p>
<ul>
<li>If you have previously registered on this server just log in:<ul>
<li>On the top menu select: <strong>User -&gt; Login</strong></li>
<li>Enter your password</li>
<li>Click the <strong>Submit</strong> button</li>
</ul>
</li>
</ul>
<!-- -->

<ul>
<li>If you haven’t registered on this server, you’ll need to now.<ul>
<li>On the top menu select: <strong>User -&gt; Register</strong></li>
<li>Enter your email, choose a password, repeat it and add a (all
    lower case) one word name</li>
<li>Click on the <strong>Submit</strong> button.</li>
</ul>
</li>
</ul>
<hr />
<h2 id="section-2-create-and-run-a-workflow">Section 2: Create and run a workflow.</h2>
<p>This section will show you two different methods to create a workflow
and then how to run one.</p>
<h3 id="import-the-workflow-history">Import the workflow history</h3>
<p>This step will show you how to import a history from a remote source
into your own workspace. We will be using this history to build a
workflow.</p>
<ul>
<li>From the histories menu
    <img alt="" src="./images//Galaxy-menu.png" />,
    click <strong>Import from File</strong>.</li>
<li>Enter
    <em>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/Workflow-finished-history.tar.gz</em>
    in the URL box.</li>
<li>Click <strong>Submit</strong></li>
</ul>
<p>You might have to wait for a bit, then:</p>
<ul>
<li>From the history menu
    <img alt="" src="./images//Galaxy-menu.png" />,
    click <strong>Saved Histories</strong></li>
<li>Select the history: <em>Imported: Workflow-finished</em></li>
</ul>
<h3 id="workflow-creation-method-1">Workflow creation: Method 1</h3>
<p>We will create a workflow from an existing history. You can use this
method to make a re-useable analysis from one you’ve already done. i.e.
You can perform the analysis once and then create a workflow out of it
to re-use it on more/new data. We will create a workflow from the
history you imported in step 1 above. The footnote below explains the
steps that created this history. These are the steps we will mimic in
the workflow.</p>
<p>Make sure your current history is the one you imported in Section 2 -
Step 1 above (<em>imported: Workflow_finished.</em>) If not, switch to it.</p>
<p><strong>Now we will create the workflow.</strong></p>
<ul>
<li>Click on the histories menu button
    <img alt="" src="./images//Galaxy-menu.png" />
    at the top of the history pane.</li>
<li>Click <strong>Extract Workflow</strong></li>
</ul>
<p>You will now be shown a page which contains the steps used to create the
history you are extracting from. We use this page to say what to include
in the workflow. We want everything here so we’ll just accept the
defaults and:</p>
<ul>
<li>Change the Workflow name to something sensible like “Basic Variant
    Calling Workflow”</li>
<li>Click <strong>Create Workflow</strong></li>
</ul>
<p>The workflow is now accessible via the bottom of the tool pane by
clicking on All Workflows.</p>
<h4 id="some-discussion">Some discussion:</h4>
<p>Have a look at your workflow. Click on its button in the workflow list.
It’s tool interface will appear. You can now run this workflow any time
you like with different input datasets. NOTE: The input data sets must
be of the same types as the original ones. i.e. Two fastq reads files
and one fasta reference sequence.</p>
<p>More interesting though is to:</p>
<ul>
<li>Click on the <strong>Workflows</strong> link in the top menu</li>
<li>Click on the down arrow on your workflow’s button.</li>
<li>Click <strong>Edit</strong></li>
</ul>
<p>A visualisation of your workflow will appear. Note the connections and
the steps.</p>
<p>Next we’ll go through how to create this workflow using the editor..</p>
<h3 id="workflow-creation-method-2">Workflow Creation: Method 2</h3>
<p>We will now create the same read mapping/variant calling workflow using
the editor directly. We need to get some reads and a reference, map the
reads to the reference using BWA, run Freebayes on the BAM to call the
variants, and finally filter the resulting vcf file.</p>
<h4 id="step-1-create-a-workflow-name-and-edit-space-step-1-create-a-workflow-name-and-edit-space">Step 1: Create a workflow name and edit space. {#step-1-create-a-workflow-name-and-edit-space}</h4>
<ul>
<li>Click on <strong>Workflow</strong> in Galaxy’s menu.</li>
<li>Click on the <strong>Create New Workflow</strong> button.</li>
<li>In the "Workflow Name" text box type: <em>Variants from scratch</em></li>
<li>Click the <strong>Create</strong> button.</li>
</ul>
<p>You’ll now see a list of workflows. Your’s will be in that list.</p>
<h4 id="step-2-open-the-editor-and-place-component-tools">Step 2: Open the editor and place component tools</h4>
<ul>
<li>Click on the down arrow on your workflow’s button.</li>
<li>Select <strong>Edit</strong></li>
</ul>
<p>You’ll be presented with a blank workflow grid.</p>
<p><strong>Add three input datafiles.</strong></p>
<ul>
<li>In the Workflow control section of the tool pane, click on <strong>Inputs
    -&gt; Input dataset</strong> three times.</li>
<li>Spread them out towards the left hand side of the workflow grid by
    clicking and dragging them around.</li>
<li>For each one, change their name.<ul>
<li>Click on each input box in turn</li>
<li>In the right hand pane (where the history usually is), change
    the name to:<ul>
<li>Reference data</li>
<li>Reads 1</li>
<li>Reads 2 - respectively.</li>
</ul>
</li>
</ul>
</li>
</ul>
<p><strong>Add in the BWA mapping step.</strong></p>
<ul>
<li>Click on <strong>NGS: Mapping -&gt; Map with BWA</strong> in the tool pane. BWA
    will be added to the workflow grid.</li>
<li>In the right hand pane (where the history usually is), change the
    following parameters.<ul>
<li>Change “Will you select a reference genome from your history or
    use a built-in index?:” to <em>Use a genome from history.</em></li>
<li>Note that the BWA box on the grid changes to match
    these settings.</li>
</ul>
</li>
</ul>
<p><strong>Connect the tools together.</strong></p>
<ul>
<li>Click and drag on the output of one of the input dataset tools to
    each of the input spots in the BWA tool. (Make connections.)<ul>
<li>"Reference data" output to reference to "Use the following
    dataset as the reference sequence"</li>
<li>"Reads 1" output to "Select first set of reads"</li>
<li>"Reads 2" output to "Select second set of reads"</li>
</ul>
</li>
</ul>
<p><strong>Add in the Freebayes (Variant Calling step.)</strong></p>
<ul>
<li>Click on <strong>NGS: Variant Calling -&gt; Freebayes</strong></li>
<li>In the right hand pane, change the following:<ul>
<li>"Choose the source for the reference list:" to <em>History</em></li>
<li>Connect "Map with BWA" bam output to "Freebayes’" bam input.</li>
<li>Connect the "Reference data" output to "Freebayes’" <em>Use the
    following dataset as the reference sequence</em> input.</li>
</ul>
</li>
</ul>
<p>If you’re keen - <strong>Note: this is optional.</strong> Also change the following
parameters the right hand pane for freebayes to make it a bit more
sensible for variant calling in bacterial genomes.</p>
<ul>
<li>"Choose parameter selection level": <em>Complete list of all options</em></li>
<li>"Set population model": <em>Yes</em></li>
<li>"Set ploidy for the analysis": <em>1</em></li>
<li>"Set input filters": <em>Yes</em></li>
<li>"Exclude alignments from analysis if they have a mapping quality
    less than": <em>20</em></li>
<li>"Exclude alleles from analysis if their supporting base quality is
    less than": <em>20</em></li>
<li>"Require at least this fraction of observations … to evaluate the
    position": <em>0.9</em></li>
<li>"Require at least this count of observations .. to evaluate the
    position": <em>10</em></li>
<li>"Set population and mappability priors": <em>Yes</em></li>
<li>"Disable incorporation of prior expectations about observations":
    <em>Yes</em></li>
</ul>
<p><strong>Add in the Filter step.</strong></p>
<ul>
<li>Click on <strong>Filter and Sort - &gt; Filter</strong></li>
<li>Connect "Freebayes’" output_vcf to the "filter" input.</li>
<li>In the right hand pane, change the following:<ul>
<li>"With the following condition": <em>c6 &gt; 500</em></li>
<li>"Number of header lines to skip": <em>56</em></li>
</ul>
</li>
</ul>
<p>Phew! We’re nearly done! The only thing left is to select which workflow
outputs we want to keep in our history. Next to each output for every
tool is a star. Clicking on the stars will select those files as
workflow outputs, everything else will be hidden in the history. In this
case we only really want the BAM file and the final variants (vcf file.)
Therefore:</p>
<p><strong>Select workflow outputs.</strong></p>
<ul>
<li>Click on the star next to "Map with BWA’s" bam file output.</li>
<li>Click on the star next to "Filter’s" output vcf.</li>
</ul>
<h4 id="step-3-save-it">Step 3: Save it!</h4>
<p>Click on the
<img alt="" src="./images//Galaxy-menu.png" />
at the top of the workflow grid and select <strong>Save</strong>.</p>
<p>Congratulations. You’ve just created a Galaxy workflow.</p>
<p>Now to run it!</p>
<h3 id="running-the-workflow">Running the workflow</h3>
<p>We will now make a new history called "Test" and run the workflow on
it’s data.</p>
<h4 id="create-the-new-history">Create the new history</h4>
<ul>
<li>From the Histories Menu, select <strong>Copy Datasets</strong></li>
<li>Select the 2 x fastq files and the Ecoli .fna file.</li>
<li>Under destination history, enter <em>Test</em> into "New History Named:"</li>
<li>Click <strong>Copy History Items</strong> button</li>
<li>Click on the link to the new history in the green bar at the top of
    the screen</li>
</ul>
<h4 id="run-the-workflow">Run the workflow</h4>
<ul>
<li>On the tools pane, click <strong>All Workflows</strong></li>
<li>Select the <em>Variants from scratch</em> workflow (or whatever you
    called it.)</li>
<li>Give it the correct files.</li>
<li><em>Ecoli ... .fna</em> for Reference data</li>
<li><em>bacterial_std_err_1.fastq</em> for Reads 1</li>
<li><em>bacterial_std_err_2.fastq</em> for Reads 2</li>
<li>Click <strong>Run Workflow</strong></li>
</ul>
<p>Your workflow will now run. It will send the right files to the right
tools at the right time to the cluster (compute engine on your machine)
and wait for them to finish. Watch as they turn yellow then green in
turn.</p>
<h2 id="what-now_1">What now?</h2>
<p>Where to start? There’s so much you can do with Workflows. You can even
run them on multiple file sets from the one setup.</p>
<hr />
<p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p></div>
            
        </div>

        <footer class="col-md-12">
            <hr>
            
            <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>

        <script src="./js/jquery-1.10.2.min.js"></script>
        <script src="./js/bootstrap-3.0.3.min.js"></script>
        <script src="./js/highlight.pack.js"></script>
        <script>var base_url = '.';</script>
        <script data-main="./mkdocs/js/search.js" src="./mkdocs/js/require.js"></script>
        <script src="./js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>

    </body>
</html>

<!--
MkDocs version : 0.15.3
Build Date UTC : 2016-05-16 16:29:37.760714
-->
