<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="./img/favicon.ico">

	<title>Introduction to Galaxy</title>

        <link href="./css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="./css/font-awesome-4.5.0.css" rel="stylesheet">
        <link href="./css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="./css/highlight.css">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body class="homepage">

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            

            <!-- Main title -->
            <a class="navbar-brand" href=".">Introduction to Galaxy</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            

            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#introduction-to-galaxy">Introduction to Galaxy</a></li>
        
            <li><a href="#background">Background</a></li>
        
            <li><a href="#learning-objectives">Learning Objectives</a></li>
        
            <li><a href="#section-1-preparation">Section 1: Preparation.</a></li>
        
            <li><a href="#section-2-getting-data-into-galaxy">Section 2: Getting data into Galaxy</a></li>
        
            <li><a href="#section-3-play-with-the-tools">Section 3: Play with the tools</a></li>
        
            <li><a href="#section-4-create-workflow">Section 4: Create workflow</a></li>
        
            <li><a href="#what-now">What now?</a></li>
        
    
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<h1 id="introduction-to-galaxy">Introduction to Galaxy</h1>
<h2 id="background">Background</h2>
<p>Galaxy is a web based analysis and workflow platform designed for
biologists to analyse their own data. It comes with most of the popular
bioinformatics tools already installed and ready for use. There are many
Galaxy servers around the world and some are tailored to specific
toolsets and reference data for analysis of human genomics, microbial
genomics, proteomics etc.</p>
<p>There are some introductory slides available
<a href="https://github.com/dyusuf/galaxy_basics/raw/master/galaxy_tutorial/Galaxy_Basics.pdf">here</a>.</p>
<p>Basically, the Galaxy interface is separated into 3 parts. The tool list
on the left, the viewing pane in the middle and the analysis and data
history on the right. We will be looking at all 3 parts in this
tutorial.</p>
<p><img alt="Galaxy interface components diagram" src="./images/Galaxy_components.png" /></p>
<hr />
<h2 id="learning-objectives">Learning Objectives</h2>
<p>At the end of this tutorial you should:</p>
<ol>
<li>
<p>Be able to upload data to a Galaxy server from:</p>
<ul>
<li>
<p>A file on your local computer</p>
</li>
<li>
<p>A file on a remote datastore with an accessible URL</p>
</li>
<li>
<p>A file on a public dataset</p>
</li>
</ul>
</li>
<li>
<p>Be able to use tools in Galaxy by:</p>
<ul>
<li>
<p>Accessing the tool via the tool menu</p>
</li>
<li>
<p>Using the tool interface to run the particular tool</p>
</li>
<li>
<p>Viewing/accessing the tool output.</p>
</li>
</ul>
</li>
<li>
<p>Be able to create a simple workflow</p>
</li>
</ol>
<hr />
<h2 id="section-1-preparation">Section 1: Preparation.</h2>
<p>The purpose of this section is to get you to log into the MDC server.</p>
<ol>
<li>
<p>Go to <a href="https://galaxy.mdc-berlin.net/galaxy/">MDC galaxy production sever</a></p>
</li>
<li>
<p>Log in:</p>
<ul>
<li>Enter your MDC user name</li>
<li>Enter your password</li>
<li>Click <strong>OK</strong></li>
</ul>
</li>
</ol>
<hr />
<h2 id="section-2-getting-data-into-galaxy">Section 2: Getting data into Galaxy</h2>
<p>There are 3 main ways to get your data into Galaxy. We will use each of
these methods for 3 files and then use those 3 files for the rest
of the workshop.</p>
<ul>
<li>
<p>Start a new history for this workshop. To do this:</p>
<ul>
<li>Click on the history menu button (the
    <img alt="" src="./images/Galaxy-menu.png" /> icon)
    at the top of the Histories panel.</li>
<li>Select <strong>Create New</strong></li>
</ul>
</li>
</ul>
<p>It is important to note that Galaxy has the concept of "File Type" built
in. This means that each file stored needs to have its type described to
Galaxy as it is being made available. Examples of file types are text,
fasta, fastq, vcf, GFF, Genbank, tabular etc.</p>
<p>We will tell Galaxy what type of file each one is as we upload it.</p>
<h3 id="method-1-upload-a-file-from-your-own-computer">Method 1: Upload a file from your own computer</h3>
<p>With this method, you can get most of the files on your own computer into
Galaxy. (there is a size limit)</p>
<ul>
<li>
<p>Download the following file to your computer:
    <em>https://www.dropbox.com/s/1oryi4ynczi7yhk/Contig_stats.txt.gz?raw=1</em></p>
<ul>
<li>From the Galaxy tool panel, click on <strong>Get Data -&gt; Upload
    File</strong></li>
<li>Click the <strong>Choose local file</strong> button</li>
<li>Find and select the <em>Contig_stats.txt.gz</em> file you downloaded
    and click <strong>Open</strong></li>
<li>Set the "Type" to <em>tabular</em></li>
<li>Click the <strong>Start</strong> button</li>
<li>Once the progress bar reaches 100%, click the <strong>Close</strong> button</li>
</ul>
</li>
</ul>
<p>The file will now upload to your current history.</p>
<h3 id="method-2-upload-a-file-from-a-url">Method 2: Upload a file from a URL</h3>
<p>If a file exists on a web resource somewhere and you know its URL
(Unique resource location - a web address) you can directly load it into
Galaxy.</p>
<ul>
<li>
<p>From the tool panel, click on <strong>Get Data -&gt; Upload File</strong></p>
<ul>
<li>Click on the <strong>Paste/Fetch Data</strong> button</li>
<li>Copy and paste the following web address into the URL/Text box:
    <em>https://www.dropbox.com/s/6gjydgapnlgvvgj/Galaxy1-GSE37268_mof3.out_chr.hpeak.txt?raw=1</em></li>
<li>Set the "Type" to <em>interval</em>, the 'Genome' to <em>mm9</em></li>
<li>Click <strong>Start</strong></li>
<li>Once the progress bar has reached 100%, click <strong>Close</strong></li>
</ul>
</li>
</ul>
<h3 id="method-3-get-data-from-a-public-dataset">Method 3: Get data from a public dataset</h3>
<p>We also need a list of genes in mouse, which we can obtain from UCSC.
Galaxy has the UCSC table browser integrated as a tool,
so we don't need to download the data to our computers.</p>
<ul>
<li>
<p>From the tool panel, lick on <strong>Get Data -&gt; UCSC main table browser</strong></p>
</li>
<li>
<p>Select clade "Mammal", genome "Mouse", assembly "mm9"</p>
</li>
<li>Select group "Genes and Gene Prediction Tracks", track "RefSeq Genes"</li>
<li>Select table "refGene"</li>
<li>Select region "genome"</li>
<li>Select output format "BED"</li>
<li>Click button "get output"</li>
<li>Click button "Send query to Galaxy"</li>
</ul>
<hr />
<h2 id="section-3-play-with-the-tools">Section 3: Play with the tools</h2>
<p>The purpose of this section is to get you used to use the available
tools in Galaxy and point out some of the more basic manipulation tools.</p>
<p>Firstly, however, you’ll notice that one file has very long and
confusing name. So we might want to change it. To do this we need to
“edit” the file. So:</p>
<ol>
<li>Click on the
    the pencil icon (edit) next to the file in the history called:
    <em>https://www.dropbox.com/s/6gjydgapnlgvvgj/Galaxy1-GSE37268_mof3.out_chr.hpeak.txt?raw=1</em></li>
<li>In the "Name" text box, give it a new name. Call it: <em>Peak interval</em></li>
<li>Click the <strong>Save</strong> button.</li>
</ol>
<p>Now that’s better. There was a lot of other functionality hidden behind
that edit
(<img alt="" src="./images/Galaxy-edit.png" />)
icon. You can change a file’s data type, convert its format and many
other things. Feel free to play around with them.</p>
<p>Ok, back to the tools.</p>
<p>This example shows how to cut out a couple of
columns, remove a line and then produce a histogram. This will introduce
some of the text manipulation tools.</p>
<p>Click on the
<img alt="" src="./images/Galaxy-view.png" />
icon of the <em>Contig_stats.txt</em> file to have a look at it. Note that
there are 18 columns in this file. We want column 1 and column 6. To do
this:</p>
<p><strong>1. Cut out column 1 and column 6.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Text Manipulation -&gt; Cut</strong> and
    set the following:</li>
<li>Set "Cut Columns" to: <em>c1,c6</em></li>
<li>"Delimited by": <em>Tab</em></li>
<li>"Cut from": <em>Contig_stats.txt</em></li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>Examine the new file by clicking on it’s
<img alt="" src="./images/Galaxy-view.png" />
icon. We now have 2 columns instead of the 18 in the original file.</p>
<p><strong>2. Remove the Header lines of the new file.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Text Manipulation -&gt; Remove
    beginning</strong> and set the following:</li>
<li>"Remove First": <em>1</em></li>
<li>"from": <em>Cut on data X</em></li>
<li>click <strong>Execute</strong></li>
</ul>
<p>Note the new file is the same as the previous one without the header
line.</p>
<p><strong>3: Visualize Genomic Intervals</strong></p>
<p>To visualize the peaks it's best to convert them to BED format first
(because interval format just exists in Galaxy).</p>
<ul>
<li>Click on the pencil icon of the dataset, <em>Peak interval</em></li>
<li>Under the header "Convert Format" select "Convert Genomic Intervals to BED"</li>
<li>Click "Convert"</li>
<li>Look at the new dataset. Some columns with generic names have been added and others were removed to comply with BED format rules.</li>
<li>Expand the converted BED dataset in history panel</li>
<li>Click on "display at UCSC main"</li>
</ul>
<hr />
<h2 id="section-4-create-workflow">Section 4: Create workflow</h2>
<p>We will create a workflow from an existing history. You can use this
method to make a re-useable analysis from one you’ve already done. i.e.
You can perform the analysis once and then create a workflow out of it
to re-use it on more/new data.</p>
<p><strong>Step 1: Find overlaps</strong></p>
<ul>
<li>From the tool panel, click on <strong>Genomic Interval Operations -&gt; Intersect interval files</strong></li>
<li>Return: Overlapping Intervals of  <em>UCSC Main on Mouse: refGene (genome)</em>
that intersect <em>peak interval</em></li>
</ul>
<p>The order of the inputs is important! We want to end up with a list of genes,
so the corresponding dataset needs to be the first input.</p>
<p><strong>Step 2: Count genes on different chromosomes</strong></p>
<p>To get a better overview of the genes we obtained, we want to look at their distribution across the different chromosomes.</p>
<ul>
<li>From the tool panel, click on <strong>Statistics -&gt; count occurrences of each record</strong></li>
<li>Input: result from step 1 (Intersect interval files on data X and data Y)</li>
<li>Select column 1 (c1) with the chromosome names</li>
<li>click <strong>Execute</strong></li>
</ul>
<p><strong>Step 3: Draw barchart</strong></p>
<p>Galaxy has a second option to visualise tabular data, with built-in dynamic visualisations:</p>
<ul>
<li>Input: result from step 2</li>
<li>Expand the dataset view and click on the visualization icon</li>
<li>Choose "Charts"</li>
<li>Enter a chart title, e.g. "Genes on different chromosomes"</li>
<li>Select "Bar diagrams" -&gt; "Regular"</li>
<li>On the top, click on "Add Data"</li>
<li>Enter a label, e.g. "count"</li>
<li>Values for x-axis: Column: 2 [str]</li>
<li>Values for y-axis: Column: 1 [int]</li>
<li>On the very top, click "Draw"</li>
</ul>
<p><strong>Step 4: Name  your history</strong></p>
<p>In the history column clicks on "Unnamed history" at the top to rename it.</p>
<p><strong>Step 5: Make a workflow out of steps 1 to 3</strong></p>
<ul>
<li>Click on the history options and select "Extract workflow"</li>
<li>select the steps you wish to include in the workflow</li>
<li>Click "Create Workflow"</li>
</ul>
<p>To make sure our workflow is correct we look at it in the editor and make some small adjustments.</p>
<p>Top menu: Workflow</p>
<ul>
<li>Click on the name of your new workflow and select "Edit"</li>
</ul>
<p>The individual steps are displayed as boxes and their outputs and inputs are connected through lines. When you click on a box you see the tool options on the right. Besides, the tools you should see two additional boxes titled "Input dataset". These represent the data we want to feed into our workflow.</p>
<p>You should also change the names of the input datasets to remember that the first one contains genes and the second one peaks. Don't forget to save it in the end by clicking on "Options" (top right) and selecting "Save".</p>
<p><strong>Step 6: Share workflow</strong>
Share your new workflow with the person to your left.</p>
<p>Top menu: Workflow</p>
<ul>
<li>Click on your workflow's name and select "Share or publish"</li>
<li>Click "Share with a user"</li>
<li>Enter the email address of the person to your left (the same as he/she uses to login to Galaxy)</li>
<li>Hit "Share"</li>
<li>Wait for the person on your right to do the same</li>
<li>Reload the workflows by clicking again on "Workflow" in the top menu</li>
<li>Under the header "Workflows shared with you by others" you should now see your right neighbor's workflow</li>
<li>Click on its name and select "View"</li>
<li>Compare with your workflow</li>
</ul>
<h3 id="running-the-workflow">Running the workflow</h3>
<p>We can now make a new history called "Test" and run the workflow on
it’s data, automating steps from 1 to 3.</p>
<h2 id="what-now">What now?</h2>
<p>Remember how we started a new history at the beginning? If you want to
see any of your old histories, click on the history menu button
<img alt="" src="./images/Galaxy-menu.png" />
at the top of the histories panel and then select “Saved Histories.”
This will give you a list of all the histories you have worked on in
this Galaxy server.</p>
<p>There’s so much you can do with Workflows. You can even
run them on multiple file sets from the one setup.</p>
<p>That's it. You now know a bit about the Galaxy interface and how to load
data, run tools and view their outputs. For more tutorials, see</p>
<ul>
<li><a href="https://github.com/bgruening/training-material">https://github.com/bgruening/training-material</a></li>
<li><a href="http://genome.edu.au/learn">http://genome.edu.au/learn</a></li>
</ul>
<hr />
<p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p></div>
            
        </div>

        <footer class="col-md-12">
            <hr>
            
            <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>

        <script src="./js/jquery-1.10.2.min.js"></script>
        <script src="./js/bootstrap-3.0.3.min.js"></script>
        <script src="./js/highlight.pack.js"></script>
        <script>var base_url = '.';</script>
        <script data-main="./mkdocs/js/search.js" src="./mkdocs/js/require.js"></script>
        <script src="./js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>

    </body>
</html>

<!--
MkDocs version : 0.15.3
Build Date UTC : 2016-08-16 19:06:50.809047
-->
